.github/workflows/upload-release-to-plugin-repo.yml

```diff
@@ -5,10 +5,66 @@ on:
         types: [published]
 
 jobs:
+    compute-should-update-trunk:
+        name: Decide if trunk or tag
+        runs-on: ubuntu-latest
+        # Skip this job if the release is a release candidate. This will in turn skip
+        # the upload jobs, which are only relevant for non-RC releases.
+        # We first check if the release is a prerelease, and then if the ref contains
+        # the string "rc". The latter is fallback in case the deployer accidentally
+        # unchecks the "This is a pre-release" checkbox in the release UI.
+        if: |
+            !github.event.release.prerelease && !contains(github.ref, 'rc')
+
+        outputs:
+            should_update_trunk: ${{ steps.compute_should_update_trunk.outputs.should_update_trunk }}
+
+        steps:
+            - name: Fetch latest version in the WP core repo
+              id: compute_latest_version_in_core_repo
+              run: |
+                  latest_version_in_core_repo=$(curl -s 'https://api.wordpress.org/plugins/info/1.2/?action=plugin_information&request\[slug\]=gutenberg' | jq -r '.version')
+                  echo "Latest Core Repo version: $latest_version_in_core_repo"
+                  echo "version=$latest_version_in_core_repo" >> $GITHUB_OUTPUT
+
+            - name: Decide if it is a trunk or tag update
+              id: compute_should_update_trunk
+              env:
+                  GITHUB_REF: ${{ github.ref }}
+              run: |
+                  latestPublishedVersion=$(echo "$GITHUB_REF" | sed -E 's/refs\/tags\/(v?)([0-9.]+)/\2/')
+                  latestVersionInCoreRepo="${{ steps.compute_latest_version_in_core_repo.outputs.version }}"
+
+                  # Determines if the first version string is greater than the second version string.
+                  #
+                  # Params:
+                  #   $1 - The first version string to compare, which may have an optional leading "v".
+                  #   $2 - The second version string to compare, which may have an optional leading "v".
+                  #
+                  # Return values:
+                  #   0 - The first version string is greater than the second version string.
+                  #   1 - The first version string is less than or equal to the second version string.
+                  is_first_version_greater_than_second() {
+                    v1=${1#v}
+                    v2=${2#v}
+                    dpkg --compare-versions "$v1" gt "$v2"
+                    return $?
+                  }
```

Comment:
Definitely not for this PR, but in a follow-up, can we maybe replace this with the [`semver` npm](https://www.npmjs.com/package/semver)? I'm already using it in a few places in the build-plugin workflow ([e.g.](https://github.com/WordPress/gutenberg/blob/1acd27e54045130bc9b3dcc0ed79a781db251056/.github/workflows/build-plugin-zip.yml#L82)) to compute version numbers, and it also supports comparison

```
semver.gt('1.2.3', '9.8.7') // false
semver.lt('1.2.3', '9.8.7') // true
```

(and cleanup: `semver.clean('  =v1.2.3   ') // '1.2.3'`).

(`dpkg` is a Debian/Ubuntu tool IIRC, and I've tried to avoid bash-isms and other dependencies as much as possible in order to make local testing of individual steps from workflows at least a bit easier across platforms -- including MacOS.)

Reply:
That's a good point. I'll take note of that for a follow-up 👍🏻